[DOC-11986] Rework the Student Record Tutorial for the Server docs (#167)
couchbaselabs/docs-devex: modules/tutorials/pages/couchbase-tutorial-student-records.adoc

= Developer Tutorial: Student Record System
:description: The Student Record System tutorial walks you through downloading and installing Couchbase, and then creating a database to store student records.
:page-topic-type: tutorial
:page-pagination: next
:page-toclevels: 2

[abstract]
{description}

== Introduction

Couchbase is a schema-less JSON document database designed for high performance, scalability, and fast development.
This tutorial teaches you about to the key concepts behind Couchbase and how they differ from traditional SQL database systems like MySQL and Oracle.

> IMPORTANT: This tutorial uses standalone or Docker installations of the Couchbase Server.

Reviewer: ```suggestion
IMPORTANT: This tutorial uses standalone or Docker installations of Couchbase Server.
```

> To get started with the Couchbase Capella Cloud service, see xref:cloud:get-started:create-account.adoc[Create a Free Account and Deploy Your Database].

Reviewer 2: Ultimately this xref should point to the same tutorial on the Capella side, so:

```suggestion
To use this tutorial with the Couchbase Capella Cloud service, see xref:cloud:tutorials:couchbase-tutorial-student-records.adoc[].
```

... but maybe save that until the Capella tutorial is ready.

student:: Information about individual students, like name and date of birth.
course:: Courses the students can take. Includes course name, faculty, and the number of credit points associated with the course.
Students can take more than one course at a time.
enrollment:: Information related to courses the students are taking.
In a relational database, this is usually a link record that creates a relationship between a student and a course.

=== Relational Model

In a relational model, the database contains a list of students and a list of courses.
Each student can enroll in multiple courses.

A student's enrollment record is stored in a separate table called `enrollment`, which links that record to the courses they're enrolling in.

[plantuml,student-record-erd,svg]
....
include::partial$diagrams/student-record-erd.puml[]
....

The `enrollment` table highlights a challenge with the relational model: each table is based on a fixed schema that only supports a single data object type, which means you cannot store a student in the same table as their enrollment record.

=== Document Model

Couchbase uses a document model that stores each record as a JSON document.
The document model:

> * Contains simple scalar types and complex types like nested records and arrays

Reviewer: ```suggestion
* Contains simple scalar types and complex types, like nested records and arrays
```

}
----

Hilary's enrollment is stored in the same document as her student details, which means child information is stored with its parent.
This structure lets you access and retrieve all of Hilary's details with one search and without the need for complex table joins.

NOTE: You should not store a student with their course record as it can lead to data duplication and make it difficult to maintain your data.
For example, you would need to access every single student record in your database to change the `credit-points`.

== Next Steps

To complete this tutorial, follow these steps:

[horizontal]

> Step 1:: xref:tutorial-install-server.adoc[Install the Couchbase Server]

Reviewer: ```suggestion
Step 1:: xref:tutorial-install-server.adoc[Install Couchbase Server]
```
